proposals/manual-bundling.md

```diff
@@ -0,0 +1,150 @@
+# Allow developers to bundle Cordova JS files manually
+
+## Goal
+
+Make Cordova integrate better with a modern JavaScript setup.
+
+Basically it comes down to that Cordova should allow using node modules. This allows better integration with tooling such as:
+
+- Package managers ([npm] / [yarn])
+- Module bundlers ([webpack] / [browserify] / [rollup])
+- IDEs (Atom ([autocomplete-modules], [js-hyperclick]))
+- Linters ([eslint-plugin-import])
+- Miscellaneous tooling ([depcheck])
+
+
+## Current situation
+
+Let's say I would like to build a Cordova app for Android using the camera plugin.
+
+Steps:
+
+1. Create a Cordova project using `cordova create`.
+2. Remove the legacy hooks directory and obsolete .npmignore.
+3. Add the desired Android platform and camera plugin.
+
+   Now this is where things get weird.
+
+   A directory `/platforms/android/platform_www` is created.
+
+   Inside this folder is a file `cordova.js`, which contains a UMD bundle with the following modules:
+
+   - `'cordova'` → https://github.com/apache/cordova-js/blob/master/src/cordova.js
+   - `'cordova/exec'` → https://github.com/apache/cordova-android/blob/master/cordova-js-src/exec.js
+   - `'cordova/platform'` → https://github.com/apache/cordova-android/blob/master/cordova-js-src/platform.js
+   - `'cordova/android/*'` → https://github.com/apache/cordova-android/tree/master/cordova-js-src/android
+   - `'cordova/plugin/android/*'` → https://github.com/apache/cordova-android/tree/master/cordova-js-src/plugin/android
+   - `'cordova/*'` → https://github.com/apache/cordova-js/tree/master/src/common (except for exec and platform, not sure if they're the `_b` postfixed ones, but it doesn't really matter)
+
+   In addition there is a `cordova_plugins.js`, which maps the plugins to a file location.
+
+   Also there is a `plugins/cordova-plugin-camera/www/*`, which contains the files from https://github.com/apache/cordova-plugin-camera/tree/master/www, but with a `.` as a path separator, whereas in any other situation this would be a `/`.
+
+   However, if I take a look at what the `cordova` package on npm is, it's the CLI, which is totally unrelated to the UMD bundle.
+   
+   In addition to `/platforms`, there's also a `/plugins` directory. This contains the following:
+   
+   - Copies of the plugins that already exist in `node_modules`, but with the dotfiles missing.
+   - `android.json` — This appears to duplicate some info from `package.json` and `config.xml`, but in another format.
+   - `fetch.json` — This appears to duplicate some info from `package.json`, but in yet another format.
+
+
+## Proposal
+
+As a modern JavaScript developer, I would like to use *<this year's popular tool>* to bundle my dependencies and code.
+
+- Get rid of `cordova.js`.
+- Get rid of `www/`.
+- Get rid of `plugins/`.
+
+### 1. Load plugins from the `node_modules` directory, not `plugins`
```

Comment:
An issue for this has already been filed https://issues.apache.org/jira/browse/CB-13059. Also it is proposed in the cordova@8 proposal https://github.com/cordova/cordova-discuss/pull/72